packages/go-musicfox/build.sh

```diff
@@ -0,0 +1,38 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/go-musicfox/go-musicfox
+TERMUX_PKG_DESCRIPTION="A netease music player in terminal."
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@anhoder"
+TERMUX_PKG_VERSION="4.4.0"
+TERMUX_PKG_DEPENDS="libc++, libflac, libvorbis"
+TERMUX_PKG_SRCURL=git+https://github.com/go-musicfox/go-musicfox
```

Comment:
Unless there is a specific reason to build from the git repo, please use the release tarball instead as it vastly simplifies the build process.
```suggestion
TERMUX_PKG_SRCURL=https://github.com/go-musicfox/go-musicfox/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
```

Reply:
In the case of using the release tarball, do I need to create a new PR to maintain the `TERMUX_PKG_VERSION` and `TERMUX_PKG_SHA256` if a new version is released?

Reply:
In theory yes. On practice we can update the package automatically with package updater. It can update packages if there is new version and no new patches are needed.

Reply:
The SHA256 sum is updated by `./scripts/bin/update-checksum` as part of the autoupdate procedure.
https://github.com/termux/termux-packages/blob/b99ff2b4d356d9c1d54d56f9aae64b9e5bc5f88a/scripts/bin/update-checksum#L58-L77

Reply:
That's great

Reply:
Unfortunately, go-musicfox's build script uses git to get version information (latest tag, commit id), can we get they from any environment variable?

Reply:
`TERMUX_PKG_VERSION` should be a suitable substitute for the latest tag, since it is extracted from it.

As for the commit hash, that might require some work, since the tarball wouldn't contain it.
But it should be fairly easy to extract it from `git ls-remote --tags https://github.com/go-musicfox/go-musicfox` by matching the tag with the value of `TERMUX_PKG_VERSION`
![image](https://github.com/termux/termux-packages/assets/43716232/8d96c4bd-b048-4524-8600-735f509e2400)
I'm not sure if `git` is implicitly available during the build process, or if it would need to be specified in `TERMUX_PKG_BUILD_DEPENDS`.

Reply:
Come to think of it...
Since the tags get output in chronological order
```bash
termux_step_post_get_source() {
	local _commit="$(git ls-remote --tags https://github.com/go-musicfox/go-musicfox \
	| tail -n1 \
	| awk '{print $1}')"
}
```
this should work to get the commit of the latest tag.

Reply:
Thanks for your help.
I'm going to do this work in build script of go-musicfox, and I'll update this pr after go-musicfox releases the next version.
Anyway, thanks again for your reply.